editor/src/clj/editor/hot_reload.clj

```diff
@@ -40,12 +41,16 @@
             remote-etag (first (get headers "If-None-Match"))
             is-cached (when remote-etag (= etag remote-etag))
             file (io/file full-path)
-            content-length (.length file) ; Returns zero if not found.
-            content (when (not is-cached)
+            content-length (if is-cached
+                             ;; The engine does not support a cached response
+                             ;; that contains a content-length other than zero.
+                             0
+                             ;; Returns zero if not found.
+                             (.length file))
+            content (when (and (= method "GET") (not is-cached))
                       (try
-                        (with-open [is (io/input-stream file)]
-                          (IOUtils/toByteArray is))
-                        (catch FileNotFoundException _
+                        (Files/readAllBytes (Paths/get (.toURI file)))
```

Comment:
You could use `(.toPath file)` instead of `(Paths/get (.toURI file))` to simplify this a bit.